services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/AutoProvisioning.java

```diff
@@ -174,33 +169,36 @@ private AutoProvisioning() {
+                                "auto-provisioning failed: device could not be created"));
                     }
 
                     // 2. set the certificate credential
-                    final var certCredential = X509CertificateCredential.fromSubjectDn(authId, List.of(new X509CertificateSecret()));
+                    final var certCredential = X509CertificateCredential.fromSubjectDn(authId,
+                            List.of(new X509CertificateSecret()));
                     certCredential.setEnabled(true).setComment(comment);
 
                     final String deviceId = r.getPayload().getId();
                     TracingHelper.TAG_DEVICE_ID.set(span, deviceId);
 
-                    return credentialsManagementService.updateCredentials(tenantId, deviceId, List.of(certCredential), Optional.empty(), span)
+                    return credentialsManagementService
+                            .updateCredentials(tenantId, deviceId, List.of(certCredential), Optional.empty(), span)
                             .compose(v -> {
                                 if (v.isError()) {
-                                    LOG.warn("auto-provisioning failed: credentials could not be set [tenant: {}, auth-id: {}, status: {}]",
-                                            tenantId, authId, v.getStatus());
-                                    return deviceManagementService.deleteDevice(tenantId, deviceId, Optional.empty(), span)
-                                            .map(OperationResult.ok(v.getStatus(),
-                                                    "Auto-provisioning failed: credentials could not be set for device ["
-                                                            + deviceId + "]",
-                                                    Optional.empty(),
-                                                    Optional.empty()));
+                                    LOG.warn("auto-provisioning failed: credentials could not be set [tenant: {}, device: {}, auth-id: {}, status: {}]",
+                                            tenantId, deviceId, authId, v.getStatus());
+                                    return deviceManagementService
+                                            .deleteDevice(tenantId, deviceId, Optional.empty(), span)
+                                            .map(getCredentialsResult(v.getStatus(),
+                                                    "auto-provisioning failed: credentials could not be set for device ["
+                                                            + deviceId + "]"));
                                 } else {
                                     span.log("auto-provisioning successful");
-                                    return Future
-                                            .succeededFuture(OperationResult.empty(HttpURLConnection.HTTP_CREATED));
+                                    if (LOG.isTraceEnabled()) {
```

Comment:
FMPOV there is no need to guard the log statement as it doesn't do anything expensive, does it?